src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -492,6 +492,26 @@ parseNumberBinder = NumberBinder <$> (sign <*> number)
          <|> (symbol' "+" >> return id)
          <|> return id
 
+parseTypedBinder :: TokenParser Binder
+parseTypedBinder = do
+  b <- typeableBinder
+  indented *> doubleColon
+  t <- parseType
+  return $ TypedBinder t b
+    where typeableBinder = P.choice (map P.try
```

Comment:
Actually, this should probably be implemented using either `fold` or `augment` in `Language.PureScript.Parser.Common`, which will match any binder, and then optionally match some other thing one or more times.